### PR TITLE
Replacing var with let/const, somewhere left var because scope, somew…

### PR DESCRIPTION
**Pull request by salesh**
#947 
Replacing var with let/const, somewhere left var because scope somewhere left var because "for" + resolved all standard issues

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
Benchmarks seems ok.

File: benchmarks/utils/runbench.js [hunk 1]
```diff
@@ -90,7 +90,8 @@ function runBenchmark (name, done) {
 }
 
 function sum (arr) {
-  var result = 0
+  let result = 0
+  /* eslint no-var: off */
```
Comment (jsumners) [inline benchmarks/utils/runbench.js:94]: If we were to directly use `eslint` with https://github.com/standard/eslint-config-standard instead of the straight `standard` package, could we update our `.eslintrc` to include this rule? I really despise having these comments scattered all over the code just to tweak a development tool. I understand that sometimes they are unavoidable (e.g. `/* istanbul skip else */`), but the fewer we have to have the better.

In general, I'm really annoyed by the latest couple releases of `standard` imposing some really drastic rules. But this "no more var" one is asinine and dangerous.

File: browser.js [hunk 2]
```diff
@@ -1,11 +1,11 @@
 'use strict'
 
-var format = require('quick-format-unescaped')
+const format = require('quick-format-unescaped')
```
Comment (jsumners) [inline browser.js:3]: @davidmarkclements are there browsers we support that do _not_ support `let` and `const`?

File: lib/levels.js [hunk 3]
```diff
@@ -55,7 +55,8 @@ function genLsCache (instance) {
   const formatter = instance[formattersSym].level
   const { labels } = instance.levels
   const cache = {}
-  for (const label in labels) {
+  /* eslint no-var: off */
+  for (var label in labels) {
```
Comment (jsumners) [inline lib/levels.js:59]: It's `for (let x = 0; x < 10; x += 1)` that is slower. I believe in this case we can do:

```js
for (const label in labels)
```

Correct, @mcollina ?

File: pino.js [hunk 4]
```diff
@@ -138,10 +138,14 @@ function pino (...args) {
     [stringifySym]: stringify,
     [formattersSym]: allFormatters
   })
-  const chindings = base === null ? '' : (name === undefined)
-    ? coreChindings(base) : coreChindings(Object.assign({}, base, { name }))
+  const chindings = base === null
+    ? ''
+    : (name === undefined)
+        ? coreChindings(base)
+        : coreChindings(Object.assign({}, base, { name }))
```
Comment (jsumners) [inline pino.js:145]: Ugh. I didn't realize we had nested ternaries in this code. I'd rather eliminate the nesting. The indentation helps, but it's still difficult to reason through.

File: lib/levels.js (hunk 3 above)
Comment (mcollina) [reply lib/levels.js:59]: Yes, correct.

Comment (jsumners) [reply lib/levels.js:59]: So everywhere we have a `for...in` in this code, please remove the eslint comment and adjust to a `for(const .. in ..)`.

File: browser.js (hunk 2 above)
Comment (mcollina) [reply browser.js:3]: I would land this and see if anybody complains. IE is finally out of the door, so I would not bother.

File: benchmarks/utils/runbench.js (hunk 1 above)
Comment (salesh) [reply benchmarks/utils/runbench.js:94]: @jsumners tried now

`{
    "extends": "standard",
    "rules": {
        "no-var": "off"
    }
}`

and 

`{
    "extends": "standard",
    "rules": {
        "no-var": 0
    }
}`

But none is working, Am I missing something, did they change something in standard 16? Even tried .eslintrc.json

Comment (jsumners) [reply benchmarks/utils/runbench.js:94]: Yes, we would have to remove the `standard` package and add the `eslint` package. Then update the `package.json` scripts accordingly.

File: pino.js (hunk 4 above)
Comment (salesh) [reply pino.js]: Done

File: lib/levels.js (hunk 3 above)
Comment (salesh) [reply lib/levels.js]: Done

File: benchmarks/utils/runbench.js (hunk 1 above)
Comment (salesh) [reply benchmarks/utils/runbench.js]: Done

File: eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (jsumners) [inline eslintrc.json:1]: Why was this renamed? Doesn't `eslint` read `.eslintrc`?

File: eslintrc.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    },
+    "extends": [
+        "standard"
+    ],
+    "parserOptions": {
+        "ecmaVersion": 12
+    },
+    "rules": {
+        "no-var": "off"
+    }
```
Comment (jsumners) [inline eslintrc.json:16]: Do we really need anything other than the `"extends"` and `"rules"` sections?

File: package.json [hunk 7]
```diff
@@ -63,6 +64,11 @@
     "bole": "^4.0.0",
     "bunyan": "^1.8.14",
     "docsify-cli": "^4.4.1",
+    "eslint": "^7.17.0",
+    "eslint-config-standard": "^16.0.2",
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-node": "^11.1.0",
+    "eslint-plugin-promise": "^4.2.1",
```
Comment (jsumners) [inline package.json:71]: Are these necessary?

File: eslintrc.json (hunk 5 above)
Comment (salesh) [reply eslintrc.json:1]: https://eslint.org/docs/user-guide/migrating-to-7.0.0#runtime-deprecation-warnings-for-eslintrc-config-files

Comment (salesh) [reply eslintrc.json:1]: There are few more options but json is one of suggested

File: package.json (hunk 7 above)
Comment (salesh) [reply package.json:71]: Didn't know if you want extra funct. I will remove

File: eslintrc.json (hunk 6 above)
Comment (salesh) [reply eslintrc.json:16]: Didn't know if you want extra funct. I will remove

File: package.json (hunk 7 above)
Comment (salesh) [reply package.json:71]: @jsumners I did an investigation and all of these 3 need to be in devDependencies - won't work without it, this package is dependent on them.

File: eslintrc.json (hunk 6 above)
Comment (salesh) [reply eslintrc.json:16]: @jsumners ESLint needs parserOptions to know which one should use - **ESLint allows you to specify the JavaScript language options you want to support. By default, ESLint expects ECMAScript 5 syntax** should I leave  "ecmaVersion": 12 ?. Do we need any "global variables that are predefined" https://eslint.org/docs/4.0.0/user-guide/configuring#specifying-environments  or you want me to remove them?

File: eslintrc.json (hunk 5 above)
Comment (jsumners) [reply eslintrc.json:1]: That link explicitly says to add a `.eslintrc` file to the project. By default, `eslint` will look for `.eslintrc` in the project directory (the parent directory when running the tool). So let's keep it `.eslintrc` and remove the `--config` in the npm script.

File: eslintrc.json (hunk 6 above)
Comment (jsumners) [reply eslintrc.json:16]: @mcollina what is your opinion of this configuration? It eliminated needing to scatter inline tool directives throughout the source code, but it is more scaffolding than just "install `standard`". Unfortunately, `standard` explicitly ignores extra configuration by design. So we can't disable this rule while only using the `standard` package.

Comment (mcollina) [reply eslintrc.json:16]: I don't really understand why we are moving away from standard. I think it was ok.

Comment (jsumners) [reply eslintrc.json:16]: We still use the standard rule set, but gain the ability to disable its `var` rule in one place instead of having to do it on every line where we want to keep using `var`.

Comment (mcollina) [reply eslintrc.json:16]: Ok!

**Review by mcollina — APPROVED** [review]
lgtm

File: eslintrc.json (hunk 6 above)
Comment (jsumners) [reply eslintrc.json:16]: Alrighty. @salesh let's rename the rc back to what it was and update the npm script. Then I'm good with this PR.

**Review by jsumners — APPROVED** [review]
LGTM
